django/db/models/options.py

```diff
@@ -250,7 +251,9 @@ def _prepare(self, model):
                 field.primary_key = True
                 self.setup_pk(field)
             else:
-                auto = AutoField(verbose_name='ID', primary_key=True, auto_created=True)
+                pk_setting = getattr(self.app_config, "model_default_pk", settings.MODEL_DEFAULT_PK)
+                pk_class = import_string(pk_setting)
+                auto = pk_class(verbose_name='ID', primary_key=True, auto_created=True)
```

Comment:
🐛 I think there's a soft bug here. Right now `model_default_pk` is a property that already defaults to `settings.MODEL_DEFAULT_PK`. In this case the third argument in the `getattr(...)` call will never be used right?

Reply:
There is one exclusion to this: the `Migration` model itself. `self.app_config` is None.